ktlint-ruleset-standard/src/test/kotlin/com/pinterest/ktlint/ruleset/standard/rules/WrappingRuleTest.kt

```diff
@@ -1871,6 +1871,303 @@ internal class WrappingRuleTest {
+                }
+
+                while (System.currentTimeMillis() % 2 == 0L) {
+                    println(System.currentTimeMillis())
+                };
+                while (Random(System.currentTimeMillis()).nextBoolean()) {
+                    println(System.currentTimeMillis())
+                }
+
+                while (System.currentTimeMillis() % 2 == 0L) {
+                    println(System.currentTimeMillis())
+                };
+                do {
+                    println(System.currentTimeMillis())
+                } while (System.currentTimeMillis() % 2 == 0L)
+            }
+            """.trimIndent()
+        wrappingRuleAssertThat(code)
+            .hasLintViolations(
+                LintViolation(4, 7, "Missing newline after \";\""),
+                LintViolation(10, 7, "Missing newline after \";\""),
+                LintViolation(16, 7, "Missing newline after \";\""),
+            )
+            .isFormattedAs(formattedCode)
+    }
+
+    @Test
+    fun `Issue 1078 - Given a multiline semi separated control flow with no body`() {
+        val code =
+            """
+            fun test() {
+                for (i in 0..10); for (i in 0..100);while (System.currentTimeMillis() % 2 == 0L); while (Random(System.currentTimeMillis()).nextBoolean());
+            }
+            """.trimIndent()
+        val formattedCode =
+            """
+            fun test() {
+                for (i in 0..10);
+                for (i in 0..100);
+                while (System.currentTimeMillis() % 2 == 0L);
+                while (Random(System.currentTimeMillis()).nextBoolean());
+            }
+            """.trimIndent()
+        wrappingRuleAssertThat(code)
+            .hasLintViolations(
+                LintViolation(2, 22, "Missing newline after \";\""),
+                LintViolation(2, 41, "Missing newline after \";\""),
+                LintViolation(2, 86, "Missing newline after \";\""),
+            )
+            .isFormattedAs(formattedCode)
+    }
+
+    @Test
+    fun `Issue 1078 - Given a multiline semi separated import statement then wrap each expression to a new line`() {
+        val code =
+            """
+            import java.util.ArrayList; import java.util.HashMap
+            """.trimIndent()
+        val formattedCode =
+            """
+            import java.util.ArrayList;
+            import java.util.HashMap
+            """.trimIndent()
+        wrappingRuleAssertThat(code)
+            .hasLintViolation(1, 28, "Missing newline after \";\"")
+            .isFormattedAs(formattedCode)
+    }
+
+    @Test
+    fun `Issue 1078 - Given a multiline semi separated with multiple items`() {
+        val code =
+            """
+            fun test() {
+                val a = 0; val b = 0; fun bar() {
+                    // no-op
+                }; for(i in 0..10) {
+                    println(i); println(i); a++; println(a)
+                }
+            }
```

Comment:
What is tested in this method that was not tested in the other tests?

Reply:
More than one statement. But anyway I will look again

Reply:
I have updated the TC name to be more clear here. TC name is `Given a multiline semi separated with variables, flow controls and method calls`